### PR TITLE
Update docs to reflect the changes made in #14789 

### PR DESCRIPTION
**Pull request by klakho0400**
This PR fixes the documentation about default record comparison.
#14789 updated those to have a lexicographical ordering. But didn't
change the documentation.

`make docs` generates doc successfully.

### CODE REVIEW

**Comment by klakho0400** [comment]
@e-kayrakli please do make the docs and check. Thank you!

File: doc/rst/language/spec/records.rst [hunk 1]
```diff
@@ -612,9 +612,8 @@ Calling ``delete`` on a record is illegal.
 Default Comparison Operators
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-For records, the compiler will supply default comparison operators if
-they are not supplied by the user. In contrast, the user cannot redefine
-``==`` and ``!=`` for classes. The default comparison operators for a
-record examine the arguments’ fields, while the comparison operators for
-classes check whether the l.h.s. and r.h.s. refer to the same class
-instance or are both ``nil``.
+"Record comparisons have a similar behaviour to tuple comparisons (https://chapel-lang.org/docs/master/language/spec/tuples.html#relational-operators).
```
Comment (e-kayrakli) [inline doc/rst/language/spec/records.rst]:
```suggestion
Record comparisons have a similar behavior to :ref:`tuple comparisons <_Tuple_Relational_Operators>`.
```

See `doc/rst/language/spec/tuples.rst` for that label to remove the open link in the doc. See `modules/internal/String.chpl:72` for an instance of these kind of references.

Also let's make it American English (behavior)

File: doc/rst/language/spec/records.rst [hunk 2]
```diff
@@ -612,9 +612,8 @@ Calling ``delete`` on a record is illegal.
...
+The operators >, >=, <, and <= check the corresponding lexicographical order based 
+on pair-wise comparisons between the arguments' fields components.
```
Comment (e-kayrakli) [inline doc/rst/language/spec/records.rst]: Drop "components"

**Comment by klakho0400** [comment]
@e-kayrakli I have made the two changes.

File: doc/rst/language/spec/records.rst [hunk 3]
```diff
@@ -612,9 +612,8 @@ Calling ``delete`` on a record is illegal.
 Default Comparison Operators
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-For records, the compiler will supply default comparison operators if
-they are not supplied by the user. In contrast, the user cannot redefine
-``==`` and ``!=`` for classes. The default comparison operators for a
-record examine the arguments’ fields, while the comparison operators for
-classes check whether the l.h.s. and r.h.s. refer to the same class
-instance or are both ``nil``.
+"Record comparisons have a similar behavior to tuple comparisons :ref:`tuple comparisons <_Tuple_Relational_Operators>`.
```
Comment (e-kayrakli) [inline doc/rst/language/spec/records.rst]: No leading underscore in the reference

**Comment by e-kayrakli** [comment]
Thanks for the quick fix. See the failures in travis, it is not able to build docs. And it is because I wasn't careful in my suggestion for the link. You don't need the leading underscore when you are adding a reference. That's my bad :(

If you want to check whether the doc builds fine locally before you push it, you do `make chpldoc` followed by `make doc`. This should generate the docs in `doc/html`

**Comment by e-kayrakli** [comment]
Also please update the PR message with a short description of what this is about (one sentence, really) because the PR message turns into the commit message that will live with this merge forever :) So it should describe what this PR does. And ideally if you run `make docs` locally and haven't seen any errors, you can add that there too. See other closed PRs or `git log --merges` for messages for already-merged PRs.

**Comment by klakho0400** [comment]
when I ran make doc I got a message Nothing to be done for doc

**Comment by e-kayrakli** [comment]
`make doc`? it should be `make docs`

**Comment by klakho0400** [comment]
make docs gave an error in error docs and docs

**Comment by klakho0400** [comment]
@e-kayrakli I think it is done

**Comment by e-kayrakli** [comment]
OK checking out and building the docs myself, I realize that you forgot about the part that this paragraph needs to go [here](https://github.com/chapel-lang/chapel/pull/15036/files#diff-924089f6ed09f7de60919a379ccb28c5L526-L529). So the part you change, shouldn't change because it is talking about a different thing.

See the [very first comment I made in the first PR](https://github.com/chapel-lang/chapel/pull/14844#issuecomment-584289341)

**Comment by klakho0400** [comment]
oh, shoot! I will change it right now

**Comment by e-kayrakli** [comment]
We are getting there :) Make sure to update the PR message. I think I put a sentence up there in one of my comments. It'd be good to refer to the issue where this was initially reported.

**Comment by klakho0400** [comment]
@e-kayrakli is it done?

**Comment by e-kayrakli** [comment]
Nope :)

See that I pointed out the very last paragraph in my [link above](https://github.com/chapel-lang/chapel/pull/15036/files#diff-924089f6ed09f7de60919a379ccb28c5L526-L529), also I have said that

> What we need to change to resolve #14833 is the last paragraph under [this heading](https://chapel-lang.org/docs/master/language/spec/records.html#default-comparison-operators).

As you can tell, you are removing parts that is not problematic. We only need to replace the part that is talking about the previous behavior when it comes to ordering records.

**Comment by klakho0400** [comment]
@e-kayrakli I am sorry for delaying it I have changed the last para of the header

**Comment by e-kayrakli** [comment]
OK hopefully, last set of feedback:

- Can you reset the change in `hello.good`? Something like `git checkout 08e18c4315a3ee5105c1defda2fa5e585efb27f7 -- test/release/examples/hello.good` Should do it. Make sure that when you check "Files changed" from above you only change one file and only the last paragraph.

- On the same vein, you can see that you removed an empty line right after that heading, please add that back.

**Comment by klakho0400** [comment]
@e-kayrakli finally done thanks for being patient and guiding me through my first PR

**Pull request merged**